disassemblers/ofrak_ghidra/ofrak_ghidra_test/test_additional_script_dirs.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from ofrak import Resource
+from ofrak_ghidra.ghidra_model import GhidraProject
+from ofrak_ghidra_test.assets.example_ghidra_package import example_ghidra_package
+
+
+@pytest.fixture(autouse=True)
+def ghidra_extension_components(ofrak_injector):
+    ofrak_injector.discover(example_ghidra_package)
+
+
+async def test_loading_new_ofrak_ghidra_package(hello_world_elf_resource: Resource):
+    """
+    Test that the
+    [GhidraProject][ofrak_ghidra.components.ghidra_analyzer.GhidraProject] object can
+    be successfully generated
+    """
+    await hello_world_elf_resource.identify()
+    ghidra_project = await hello_world_elf_resource.view_as(GhidraProject)
+    assert isinstance(ghidra_project, GhidraProject)
+
+    await hello_world_elf_resource.run(example_ghidra_package.GhidraExampleComponent, None)
```

Comment:
Is it worth asserting anything about the resulting analysis? 

That way this test will confirm not only that it runs successfully, but also that the information is propagated back to OFRAK.

Reply:
I don't think so, that main thing the test is addressing is that the the headless Ghidra invocation succeeds, the server starts, and the process stays alive as the server waits for connections/analysis requests. Without escaping the semicolon joining script directories, this doesn't happen, and the test fails long before it gets to run a script. The script itself and any data returned doesn't matter, as long as OFRAK can get to the point of successfully sending a request to the server within the headless instance.